configure.ac

```diff
@@ -3246,139 +3246,95 @@ AC_ARG_ENABLE(llvm,[  --enable-llvm	Enable the LLVM back-end], enable_llvm=$enab
-       llvm_api_version=`awk '/MONO_API_VERSION/ { print $3 }' $llvm_config_path`
-   fi  
-
-   if test "x$llvm_api_version" = "x"; then
-	  LLVM_CFLAGS="$LLVM_CFLAGS -DLLVM_API_VERSION=0"
-	  LLVM_CXXFLAGS="$LLVM_CXXFLAGS -DLLVM_API_VERSION=0"
-   else
-	  LLVM_CFLAGS="$LLVM_CFLAGS -DLLVM_API_VERSION=$llvm_api_version"
-	  LLVM_CXXFLAGS="$LLVM_CXXFLAGS -DLLVM_API_VERSION=$llvm_api_version"
-   fi   
+	esac
 
-   AC_DEFINE_UNQUOTED(LLVM_VERSION, "$llvm_version", [Full version of LLVM libraries])
+	LLVM_CODEGEN_LIBS="$llvm_codegen"
+	if test "x$host" != "x$target"; then
+		# No need for jit libs
+		LLVM_CODEGEN_LIBS=""
+	fi
+	AC_SUBST(LLVM_CODEGEN_LIBS)
+	# Rest of the options are filled out in mono/mini/Makefile.am.in
+else # is win32
+	LLVM_ARCH_LIBS="-lLLVMX86Disassembler -lLLVMX86CodeGen -lLLVMX86AsmParser -lLLVMX86Desc -lLLVMX86Info -lLLVMX86AsmPrinter -lLLVMX86Utils"
+	case "$target" in
+	arm*)
+		LLVM_ARCH_LIBS="-lLLVMARMDisassembler -lLLVMARMCodeGen -lLLVMARMAsmParser -lLLVMARMDesc -lLLVMARMInfo -lLLVMARMAsmPrinter"
+		;;
+	aarch64*)
+		LLVM_ARCH_LIBS="-lLLVMARMDisassembler -lLLVMARMCodeGen -lLLVMARMAsmParser -lLLVMARMDesc -lLLVMARMInfo -lLLVMARMAsmPrinter"
+		;;
+	powerpc*)
+		LLVM_ARCH_LIBS="-lLLVMPowerPCDisassembler -lLLVMPowerPCCodeGen -lLLVMPowerPCAsmParser -lLLVMPowerPCDesc -lLLVMPowerPCInfo -lLLVMPowerPCAsmPrinter"
+		;;
+	esac
 
-   AC_SUBST(LLVM_CFLAGS)
-   AC_SUBST(LLVM_CXXFLAGS)
-   AC_SUBST(LLVM_LIBS)
-   AC_SUBST(LLVM_LDFLAGS)
-   AC_DEFINE(ENABLE_LLVM, 1, [Enable the LLVM back end])
-fi
+	LLVM_CFLAGS_EXTERNAL="-I$with_llvm/include -DNDEBUG -D__NO_CTYPE_INLINE -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS"
+	LLVM_CXXFLAGS_EXTERNAL="$LLVM_CFLAGS -std=gnu++11 -fvisibility-inlines-hidden -fno-rtti -Woverloaded-virtual -Wcast-qual"
+	LLVM_LDFLAGS_EXTERNAL="-L$with_llvm/lib"
+	LLVM_SYSTEM_LIBS="-lshell32 -lpsapi -limagehlp -ldbghelp -lm"
+	LLVM_LIBS_EXTERNAL="-lLLVMLTO -lLLVMObjCARCOpts -lLLVMLinker -lLLVMipo -lLLVMVectorize -lLLVMBitWriter \
+	-lLLVMTableGen -lLLVMDebugInfo -lLLVMOption -lLLVMSelectionDAG -lLLVMAsmPrinter \
+	-lLLVMMCDisassembler -lLLVMJIT -lLLVMAnalysis -lLLVMTarget \
+	-lLLVMIRReader -lLLVMAsmParser -lLLVMLineEditor -lLLVMMCAnalysis -lLLVMInstrumentation \
+	-lLLVMInterpreter -lLLVMCodeGen -lLLVMScalarOpts -lLLVMInstCombine -lLLVMTransformUtils \
+	-lLLVMipa -lLLVMAnalysis -lLLVMProfileData -lLLVMMCJIT -lLLVMTarget -lLLVMRuntimeDyld \
+	-lLLVMObject -lLLVMMCParser -lLLVMBitReader -lLLVMExecutionEngine -lLLVMMC -lLLVMCore \
+	-lLLVMSupport -lstdc++"
+	LLVM_LIBS_EXTERNAL="$LLVM_LIBS $LLVM_ARCH_LIBS $LLVM_SYSTEM_LIBS"
+
+	AC_SUBST(LLVM_CFLAGS_EXTERNAL)
+	AC_SUBST(LLVM_CXXFLAGS_EXTERNAL)
+	AC_SUBST(LLVM_LIBS_EXTERNAL)
+	AC_SUBST(LLVM_LDFLAGS_EXTERNAL)
+fi # win32
+
+AC_DEFINE(ENABLE_LLVM, 1, [Enable the LLVM back end])
+ENABLE_LLVM=1
+AC_SUBST(ENABLE_LLVM)
+fi
+
+AM_CONDITIONAL(INTERNAL_LLVM, [test "x$internal_llvm" != "x0"])
+AC_SUBST(internal_llvm)
```

Comment:
You can remove this one because you already have `AM_CONDITIONAL(INTERNAL_LLVM, ...) at https://github.com/mono/mono/pull/8792/files#diff-67e997bcfdac55191033d57a16d1408aR3330